service/client_state.cc

```diff
@@ -254,6 +257,20 @@ future<> service::client_state::ensure_exists(const auth::resource& r) const {
     });
 }
 
+void service::client_state::ensure_alterable_permisssions(const auth::resource& r) const {
```

Comment:
I don't see why this new code should be in client_state.cc - as far as I understand (but correct me if I'm wrong), it doesn't have anything to do with the client's permissions - it's just an intrinsic feature of the function name. So perhaps you can put it in permission_altering_statement.cc?